src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -764,12 +767,44 @@ private File resolveArtifactSources( Artifact artifact )
     {
         if ( transformers == null )
         {
-            return Collections.emptyList();
+            return getDefaultResourceTransformers();
         }
 
         return Arrays.asList( transformers );
     }
 
+    private List<ResourceTransformer> getDefaultResourceTransformers()
+    {
+        final List<ResourceTransformer> transformers = new LinkedList<>();
+        if ( missTransformer( ServicesResourceTransformer.class ) )
```

Comment:
`getDefaultResourceTransformers()` is only called when transformers is `null`, so no need for `missTransformer()`-call